Modules/_elementtree.c

```diff
@@ -1422,6 +1422,20 @@ element_getitem(PyObject* self_, Py_ssize_t index)
     return Py_NewRef(self->extra->children[index]);
 }
 
+static int
+element_bool(PyObject* self_)
+{
+    ElementObject* self = (ElementObject*) self_;
+    (void)PyErr_WarnEx(PyExc_DeprecationWarning,
```

Comment:
I believe you can return -1 to indicate that an error occurred (so something like `if (PyErr_WarnEx(...) < 0) { return -1; }`. This is useful so that people who want to be strict about deprecations can turn this into an exception.